src/hr/algebra/pi/model/Terminal.java

```diff
@@ -1,81 +1,94 @@
 package hr.algebra.pi.model;
 
-import hr.algebra.pi.configuration.EmployeePayRates;
-import hr.algebra.pi.configuration.TicketPrices;
-
-import java.util.ArrayList;
+import hr.algebra.pi.factory.BigVehicleFactory;
+import hr.algebra.pi.factory.SmallVehicleFactory;
+import hr.algebra.pi.factory.VehicleFactory;
 
 public class Terminal {
-    private final Train smallTrain;
-    private final Train largeTrain;
-    private ArrayList<Employee> employees;
-    private int totalRevenue;
-
-    public Terminal() {
-        smallTrain = new Train("small", 8);
-        largeTrain = new Train("large", 6);
-        employees = new ArrayList<>();
-        totalRevenue = 0;
-    }
+    private int smallTrainCapacity = 2;
+    private int bigTrainCapacity = 6;
+    private int smallTrainOccupancy = 0;
+    private int bigTrainOccupancy = 0;
+    private double totalRevenue = 0;
+    private Employee employee1 = new Employee(0.1);
+    private Employee employee2 = new Employee(0.11);
+    private double employee1Revenue = 0;
+    private double employee2Revenue = 0;
+    private double minGasThreshold = 0.1;
+    private double minBatteryThreshold = 0.1;
 
-    public void addEmployee(Employee employee) {
-        employees.add(employee);
-    }
+    private VehicleFactory smallVehicleFactory = new SmallVehicleFactory();
+    private VehicleFactory bigVehicleFactory = new BigVehicleFactory();
 
-    public void checkInVehicle(Employee employee, Vehicle vehicle) {
-        int ticketPrice = 0;
-        if (vehicle.size.equals("small")) {
-            if (smallTrain.addVehicle(vehicle)) {
-                if (vehicle.type.equals("car")) {
-                    ticketPrice = TicketPrices.SMALL_VEHICLE_CAR_PRICE;
-                } else if (vehicle.type.equals("van")) {
-                    ticketPrice = TicketPrices.SMALL_VEHICLE_VAN_PRICE;
-                }
+    public boolean parkVehicle(String type, String size, double gasLevel, double batteryLevel) {
+        IVehicle vehicle = null;
+        if (size.equalsIgnoreCase("small")) {
```

Comment:
Be careful with long if statements, especially if they depend on the object type :)